### PR TITLE
RavenDB-22492: Implementation of a UtcNow accessor to centralize access and control calls.

### PR DESCRIPTION
**Pull request by redknightlois**
### Issue link
https://issues.hibernatingrhinos.com/issue/RavenDB-22492

### Additional description

When faced with debug information or other non time critical usages, we could avoid having to read the CPU ticks counter or the OS kernel call required to read them. In those cases we are well enough knowing that we are going to get a monotonically increasing ticks counter and values that are updated at worst every 1 second.

### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [x] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: src/Sparrow.Server/TimestampAccessor.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    // needed for accuracy. 
+    public static class TimestampAccessor
+    {
+        private static long _timestamp;
+        private static readonly Timer _timer;
+
+        // Static constructor to initialize the timestamp and timer
+        static TimestampAccessor()
+        {
+            _timestamp = DateTime.UtcNow.Ticks;
```
Comment (ayende) [inline src/Sparrow.Server/TimestampAccessor.cs:17]: This is implemented using VDSO, basically it is reading a known memory location.
On Linux, `UtcNow` is getting from:
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/DateTime.Unix.cs#L14
https://github.com/dotnet/runtime/blob/5f8a2ee126eb1b4c96d867efa6b7914839b29910/src/native/libs/System.Native/pal_datetime.c#L32

That should invoke the VDSO path, meaning that the actual call is literally reading a memory location.

On Windows, this goes to:
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/DateTime.Windows.cs

That is using KUSER_SHARED_DATA, basically the same concept as VDSO, but without the need for code injection, since that is implemented directly by kernel32.

See more here: 
https://osm.hpi.de/wrk/2007/08/getting-os-information-the-kuser_shared_data-structure/

Kestrel is doing something similar:
https://github.com/dotnet/aspnetcore/blob/main/src/Servers/Kestrel/Core/src/Internal/Http/DateHeaderValueManager.cs

But on a 1 sec scale (also formatting the date itself).


In other words, this looks very much like you are just ending up copying the value to your own memory, whereas we'll call that anyway through the normal API.

Comment (redknightlois) [reply src/Sparrow.Server/TimestampAccessor.cs:17]: True, but one memory is in the kernel and requires an interop call which costs a lot. I was able to get rid completely of the cost after doing this, which means that the overhead is actually quite high.

Comment (ayende) [reply src/Sparrow.Server/TimestampAccessor.cs:17]: There is no kernel/usermode switch here, mind.

**Comment by redknightlois** [comment]
Under risk assessment we decided to move this into the 6.2 release. We will include this PR into the feature branch instead.
https://github.com/redknightlois/ravendb/tree/arm
